hack/build-manifest.sh

```diff
@@ -26,10 +26,11 @@ export CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
 DEPLOY_OPTIONS=$1
 for opt in ${DEPLOY_OPTIONS}; do export "$opt"=true; done
 
-version=$(kubectl version --short | grep 'Client Version' | sed 's/.*v//g' | cut -b -4)
-if [ 1 -eq "$(echo "${version} < 1.21" | bc)" ]; then
-    echo "You need to update your kubectl version to 1.21+ to support kustomize"
-    exit 1
+kc_version_ok=$(kubectl version --client -ojson |
+  jq '.clientVersion | (.major | tonumber) >= 1 and (.minor | tonumber) > 21')
```

Comment:
@jiere @SamYuan1990 Do you know if this is checked in CI?

Reply:
I guess not all env has `jq` installed ?

Reply:
Yes, but we can say the same for all tools we use (including `sed`), can't we :) ? 

In those env, the script fails at which point the user should install jq. 
The way we solve this in `kepler-operator` is to provide a `make tools` target that ensures tools used by the project are installed to `<project-root>/tmp/bin` 
see: https://github.com/sustainable-computing-io/kepler-operator/blob/v1alpha1/hack/tools.sh which is what `make tools` invokes. 

I am happy to copy the script from there to our repo.

Reply:
yes, If we can do that..

sed is almost pre-installed in most of dev linux env but jq is not .. 
though the dev might find that error but if we can detect or error out it will be better

Reply:
I will start by submitting the `make tools` first and the every script can start by `validate_prerequisites` which checks if required binaries are present.